testapi.pm

```diff
@@ -303,33 +300,20 @@ sub _check_backend_response ($rsp, $check, $timeout, $mustmatch) {
                 check => $check,
         }) and return 'try_again';
 
-        if ($check) {
-            # only care for the last one
-            $failed_screens = [$final_mismatch];
-        }
+        # only care for the last one
+        $failed_screens = [$final_mismatch] if $check;
         for my $l (@$failed_screens) {
             my $img = tinycv::from_ppm(decode_base64($l->{image}));
             my $result = $check ? 'unk' : 'fail';
             $result = 'unk' if ($l != $final_mismatch);
-            if ($rsp->{saveresult}) {
-                $autotest::current_test->record_screenfail(
-                    img => $img,
-                    needles => $l->{candidates},
-                    tags => $tags,
-                    result => $result,
-                    frame => $l->{frame},
-                );
-            }
-            else {
-                $autotest::current_test->record_screenfail(
-                    img => $img,
-                    needles => $l->{candidates},
-                    tags => $tags,
-                    result => $result,
-                    overall => $check ? undef : 'fail',
-                    frame => $l->{frame},
-                );
-            }
+            $autotest::current_test->record_screenfail(
+                img => $img,
+                needles => $l->{candidates},
+                tags => $tags,
+                result => $result,
+                overall => (!$rsp->{saveresult} && $check) ? undef : 'fail',
```

Comment:
Have you checked where `overall` is used and that `undef` vs. absence doesn't make a difference?